docs/source/channels.rst

```diff
@@ -0,0 +1,253 @@
+
+.. code-block:: python
+
+    chan.stop()
+
+
+Very short-lived clients
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you wish to submit work to your cluster from a short lived client such as a
+web application view, an AWS Lambda function or some other fire and forget script,
+channels give a way to do this.
+
+Motivating Example - A workflow designer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine the following deployment scenario:
+
+.. image:: images/channels_example.png
+    :alt: Workflow designer application
+
+1) A `Custom Dask Graph`_ is designed in a web app, perhaps inserting code snippets.
+
+.. code-block:: javascript
+
+    {
+        "tracking_key_1" :["allowed_module.func", arg1, arg2],
+        "tracking_key_2" :["allowed_module.func", "tracking_key_1"]
+    }
+
+2) Import strings or code snippets are validated and imported to function objects.
+The graph might be persisted and appropriate keys added for tracking. The graph is
+added to a channel that is followed by one of the clients. As the client will
+only exist for the lifetime of the request then the graph is transmitted using scatter.
+
+.. code-block:: python
+
+    client = Client('scheduler-address:8786')
+    chan = client.channel('input')
+    dask_graph = {
+                    "tracking_key_1" :(func, arg1, arg2),
+                    "tracking_key_2" :(func2, "tracking_key_1")
+                 }
+    future_list = client.scatter([("<worflow_run_id>", dask_graph)])
+    chan.append(future_list[0])
+    chan.flush()
+
+3) A long running client process is listening for new new futures
+on the channel ``input``. A long running local client function
+``hold_completed_futures`` listens for output tasks to hold them ready
+for a different client. Dask graphs are submitted and run in the
+``run_on_remote`` function. The constituent futures are added to the output
+graph so that the python web services can view status.
+
+
+.. code-block:: python
+
+    from distributed import Client
+    from distributed.client import Future, sync
+    from distributed import local_client
+    import time
+
+    def run_on_remote(dsk):
+        with local_client() as c:
+            output_channel = c.channel("output")
+            keys = list(dsk.keys())
+            keys.sort()
+            #nonblocking get initially so consituent futures can be added to a channel
+            future = c._get(dsk, keys)
+            for key in keys:
+                future = Future(key, c)
+                output_channel.append(future)
+            #blocking get in order to ensure task is completed
+            data = c.get(dsk, keys)
+        return data
+
+    def hold_completed_futures():
+        with local_client() as c:
+            output_channel = c.channel("output")
+            for future in output_channel:
+                print("holding a future")
+                time.sleep(3600)
+                if future.status != "finished":
+                    print(future.status)
```

Comment:
Thanks so much for the tips and patience, I will try to clean this up and create a blog post in the next couple of weeks.